chore(docs): add useFragment_experimental documentation (#10099)
apollographql/apollo-client: docs/source/api/react/hooks-experimental.mdx

---
title: Hooks (experimental)
description: Apollo Client experimental react hooks API reference
---

import UseFragmentOptions from '../../../shared/useFragment-options.mdx';

## `useFragment_experimental`

### Installation

> > ⚠️ **The `useFragment_experimental` hook is currently at the [preview stage](https://www.apollographql.com/docs/resources/release-stages/#preview) in Apollo Client and is available by installing `@apollo/client@beta`.** If you have feedback on it, please let us know via [GitHub issues](https://github.com/apollographql/apollo-client/issues/new?assignees=&labels=&template=bug.md).

Reviewer: ```suggestion
> ⚠️ **The `useFragment_experimental` hook is currently at the [preview stage](/resources/release-stages/#preview) in Apollo Client and is available by installing `@apollo/client@beta`.** If you have feedback on it, please let us know via [GitHub issues](https://github.com/apollographql/apollo-client/issues/new?assignees=&labels=&template=bug.md).
```

Reviewer: Totally, absolute URLs definitely work! I _think_ there's an issue with our cross-docset link validator catching broken links if they're absolute though. I'm going to confirm tomorrow to make sure I'm not recommending things due to outdated information 😬 

In any case, definitely not a blocker for this link to use _either_ format!

Author: Interesting, thanks for the context! Will also take a look tomorrow :)

> Beginning with version `3.7.0`, Apollo Client Web has preview support for the `useFragment_experimental` hook which which represents a lightweight live binding into the `ApolloCache`, returning an always-up-to-date view of whatever data the cache currently contains for a given fragment. `useFragment_experimental` never triggers network requests of its own.

Reviewer: ```suggestion
Beginning with version `3.7.0`, Apollo Client Web has preview support for the `useFragment_experimental` hook, which which represents a lightweight live binding into the Apollo Client Cache. This hook returns an always-up-to-date view of whatever data the cache currently contains for a given fragment. `useFragment_experimental` never triggers network requests of its own.
```

> `useFragment_experimental` allows Apollo Client to broadcast very specific fragment results to individual components, though it is worth noting the `useQuery` hook remains the sole hook responsible for querying and populating data in the cache ([see the API reference](./hooks#usequery)). As a result, the component reading the fragment data via `useFragment_experimental` will still be subscribed to all changes in the query data instead of only changes to the fragment's data.

Reviewer: ```suggestion
`useFragment_experimental` enables Apollo Client to broadcast very specific fragment results to individual components. Note that the `useQuery` hook remains the sole hook responsible for querying and populating data in the cache ([see the API reference](./hooks#usequery)). As a result, the component reading the fragment data via `useFragment_experimental` is still subscribed to all changes in the query data, instead of only changes to the fragment's data.
```

### Using `useFragment_experimental`

> A [GraphQL fragment](http://graphql.org/learn/queries/#fragments) is a piece of logic that can be shared between multiple queries and mutations. [See the API reference.](../../data/fragments)

Given the following fragment definition:

```js
const ItemFragment = gql`
  fragment ItemFragment on Item {
    text
  }
`;

> ```
> We can first use the `useQuery` hook to retrieve a list of items with `id`s.

Reviewer: ````suggestion
```

We can first use the `useQuery` hook to retrieve a list of items with `id`s:
````

```jsx
const listQuery = gql`
  query {
    list {
      id
    }
  }
`;
function List() {
  const { loading, data } = useQuery(listQuery);
  return (
    <ol>
      {data!.list.map(item => <Item key={item.id} id={item.id}/>)}
    </ol>
  );
}

> ```
> We can then use `useFragment_experimental` from within the `<Item>` component to create a live binding for each item by providing the `fragment` document, `fragmentName` and object reference via `from`.

Reviewer: ````suggestion
```

We can then use `useFragment_experimental` from within the `<Item>` component to create a live binding for each item by providing the `fragment` document, `fragmentName`, and object reference via `from`:
````

```tsx
function Item(props: { id: number }) {
  const { complete, data } = useFragment_experimental({
    fragment: ItemFragment,
    fragmentName: "ItemFragment",
    from: {
      __typename: "Item",
      id: props.id,
    },
  });
  return <li>{complete ? data!.text : "incomplete"}</li>;
}
```

### `useFragment_experimental` API

Supported options and result fields for the `useFragment_experimental` hook are listed below.

Most calls to `useFragment_experimental` can omit the majority of these options, but it's useful to know they exist.

#### Options

The `useFragment_experimental` hook accepts the following options:

> <UseFragmentOptions />

Author: Followed a convention I noticed where options are documented in their own `.mdx` file

---
title: Hooks (experimental)
description: Apollo Client experimental react hooks API reference
---

import UseFragmentOptions from '../../../shared/useFragment-options.mdx';
import UseFragmentResult from '../../../shared/useFragment-result.mdx';

## `useFragment_experimental`

### Installation

> ⚠️ **The `useFragment_experimental` hook is currently at the [preview stage](https://www.apollographql.com/docs/resources/release-stages/#preview) in Apollo Client and is available by installing `@apollo/client@beta`.** If you have feedback on it, please let us know via [GitHub issues](https://github.com/apollographql/apollo-client/issues/new?assignees=&labels=&template=bug.md).

Beginning with version `3.7.0`, Apollo Client Web has preview support for the `useFragment_experimental` hook, which represents a lightweight live binding into the Apollo Client Cache. This hook returns an always-up-to-date view of whatever data the cache currently contains for a given fragment. `useFragment_experimental` never triggers network requests of its own.

> `useFragment_experimental` enables Apollo Client to broadcast very specific fragment results to individual components. Note that the `useQuery` hook remains the sole hook responsible for querying and populating data in the cache ([see the API reference](./hooks#usequery)). As a result, the component reading the fragment data via `useFragment_experimental` is still subscribed to all changes in the query data instead of only changes to the fragment's data.

Reviewer 2: ```suggestion
`useFragment_experimental` enables Apollo Client to broadcast very specific fragment results to individual components. Note that the `useQuery` hook remains the primary hook responsible for querying and populating data in the cache ([see the API reference](./hooks#usequery)). As a result, the component reading the fragment data via `useFragment_experimental` is still subscribed to all changes in the query data, but receives updates only when the fragment's data changes.
```
rewording for accuracy
